src/main/java/com/ericsson/eiffelcommons/subscriptionobject/RestPostSubscriptionObject.java

```diff
@@ -12,17 +21,50 @@
     public RestPostSubscriptionObject(String subscriptionName) throws IOException {
         super(subscriptionName);
 
-        subscriptionJson.put("notificationType", "REST_POST");
+        subscriptionJson.put(NOTIFICATION_TYPE_KEY, NOTIFICATION_TYPE);
     }
 
     /**
      * Sets the field authenticationType to BASIC_AUTH together with the username and password.
      * @param username
      * @param password
      */
+    @Deprecated
     public void setBasicAuth(String username, String password) {
-        subscriptionJson.put("userName", username);
-        subscriptionJson.put("password", password);
-        subscriptionJson.put("authenticationType", "BASIC_AUTH");
+        setAuthenticationType(BASIC_AUTH);
+        setUsername(username);
+        setPassword(password);
+    }
+
+    /**
+     * Sets the field authenticationType to given value
+     * @param basicAuth
+     * @return RestPostSubscriptionObject
+     */
+    public RestPostSubscriptionObject setAuthenticationType(String basicAuth) {
```

Comment:
This method is made to sound general but the parameter says basicAuth. Perhaps call it authType or something.

Reply:
Done